src/app/components/events/create-event/create-event.component.ts

```diff
@@ -39,42 +58,84 @@ export class CreateEventComponent implements OnInit {
       },
     );
 
-    this.eventService.getUserProfiles().subscribe(
+    this.userProfilesService.getAll().subscribe(
       (profiles) => {
         this.userProfilesList = profiles;
       },
       (error) => {
         console.error('Error fetching user profiles:', error);
       },
     );
+
+    this.groupService.getAll().subscribe(
+      (groups) => {
+        this.userGroupsList = groups;
+      },
+      (error) => {
+        console.error('Error fetching user groups:', error);
+      },
+    );
+
+    this.eventForm.get('eventTemplateName')?.valueChanges.subscribe(() => {
+      this.handleEventTemplateChange();
+    });
   }
 
   private convertBetTemplateType(type: BetTemplateType): string {
     return type.toString();
   }
 
-  private convertBetTemplatesToCompleteBetTypes() {
+  private mapEventTemplateToBetTypeList(eventTemplate: EventTemplate) {
+    return eventTemplate.betTemplates.map((betTemplate) => {
+      return {
+        name: betTemplate.name,
+        type: this.convertBetTemplateType(betTemplate.type),
+        multipleChoiceOptions: betTemplate.multipleChoiceOptions || [],
+      };
+    });
+  }
+
+  handleEventTemplateChange() {
+    if (this.eventForm.get('eventTemplateName')?.value === 'Custom') {
+      this.eventForm.patchValue({
+        completeBetTypeDtoList: [],
+      });
+      return;
+    }
+
     const selectedEventTemplate = this.eventTemplates.find(
-      (template) => template.name === this.formData.selectedTemplate,
+      (template) =>
+        template.name === this.eventForm.get('eventTemplateName')?.value,
     );
 
     if (selectedEventTemplate) {
-      this.formData.completeBetTypeDtoList =
-        selectedEventTemplate.betTemplates.map((betTemplate) => {
-          const completeBetType: CompleteBetType = {
-            name: betTemplate.name,
-            type: this.convertBetTemplateType(betTemplate.type),
-            multipleChoiceOptions: betTemplate.multipleChoiceOptions || [],
-          };
-          return completeBetType;
-        });
+      this.eventForm.patchValue({
+        completeBetTypeDtoList: this.mapEventTemplateToBetTypeList(
+          selectedEventTemplate,
+        ),
+      });
     }
+    // Allow time for the DOM to update
+    setTimeout(() => {
+      const section = document.getElementById('bet-section');
+      if (section) {
+        section.scrollIntoView({ behavior: 'smooth', block: 'start' });
+      }
+    }, 0);
```

Comment:
if you set the time to 0ms you can remove the setTimeout, because is basically does nothing

Reply:
The reason for the 0ms timeout is to give the DOM enough time to update, in order for the previously added content to exist in the HTML section . Without this timeout, the behavior of the scroll is not the one intended. ChatGPT explains this much better than I ever could: 
_The setTimeout function with a timeout of 0ms, as used in your code snippet, is a common technique in JavaScript to defer the execution of a piece of code until after the current call stack has cleared. Despite the timeout being set to 0 milliseconds, it doesn't actually execute the callback function immediately. Instead, it places the callback function on the event queue, and it will only be executed once the current execution context (including all currently executing scripts) completes, and the JavaScript runtime is idle._